onepwd/plugins/action/upload_s3_secret.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+'''
+
+EXAMPLES = """
+"""
+
+RETURN = """
+"""
+
+from subprocess import Popen, PIPE
+from ansible.errors import AnsibleError
+from ansible.module_utils._text import to_bytes, to_text
+from ansible.plugins.action import ActionBase
+import os
+import onepwd
+import url64
+
+# Provides the ability to upload a secret named 's3' to a vault (if secret does not exist), and to update the values of the secret if they differ from the requested ones (set OVERWRITE=True) 
+# How to use in Ansible: 
+# action: schulcloud.onepwd.upload_s3_secret vault=infra-dev  ACCESS_KEY=my-access-key BUCKET_NAME=my-bucket-name ACCESS_SECRET=my-access-secret OVERWRITE=True
+
+# https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#action-plugins
+class ActionModule(ActionBase):
+
+    def run(self, tmp=None, task_vars=None, **kwargs):
+        # Log into OnePassword
+        login_secret=onepwd.get_op_login()
+        session_shorthand=kwargs.get('session_shorthand', os.getenv('USER'))
+        session_timeout=kwargs.get('session_timeout', 30)
+        op = onepwd.OnePwd(secret=login_secret, shorthand=session_shorthand, session_timeout=session_timeout)
+
+        # Getting Vars from Ansible 
+        # required values
+        try: 
+            vault = self._task.args['vault']
+            BUCKET_NAME = self._task.args['BUCKET_NAME']
+            ACCESS_KEY = self._task.args['ACCESS_KEY']
+            ACCESS_SECRET = self._task.args['ACCESS_SECRET']
+        except: 
+            print("""
+            ERROR! Couldn't upload s3 secret.
+            Please provide a vault, BUCKET_NAME, ACCESS_KEY and ACCESS_SECRET!
+            OVERWRITE is optional and set to 'False' per default. Set to 'True' if you wish to overwrite. 
+            """) 
+            return PLEASE_SET_REQUIRED_VALUES
+        # optional values
+        overwrite = False
+        try: 
+            overwrite = self._task.args['OVERWRITE']
+            if overwrite in ['True', 'true', 'TRUE']:
+                overwrite = eval('True')
```

Comment:
I think there should be an error if overwrite is set to something other than true or false

Reply:
done